### PR TITLE
[stable/etcd-operator] Allow creating etcd cluster on chart install

### PR DESCRIPTION
**Pull request by arm4b**
#### What this PR does / why we need it:
`stable/etcd-operator` can't deploy etcd cluster on install. Because of that it's realistically impossible to use it in other Helm charts as dependency.
This happens because CRD `EtcdCluster` is created at a later point by operator itself and not available during first helm run:

```
helm install .
Error: validation failed: unable to recognize "": no matches for kind "EtcdCluster" in version "etcd.database.coreos.com/v1beta2"
```

This PR moves `EtcdCluster` CustomResourceDefinition creation to Helm templating side and so user could deploy etcd immediately.

#### Which issue this PR fixes
  - initially https://github.com/StackStorm/stackstorm-ha/pull/81
  - related issues/PRs #728, #754, #13582

#### Special notes for your reviewer:
It looks like previous behavior before https://github.com/helm/charts/pull/754 was to deploy etcd cluster by default, but removed due to old issues described in #728.
@lachie83 @alejandroEsc With this PR we could deploy etcd cluster by default and my proposal is to do that again.
If we stick with such defaults, this will allow me to add new Helm e2e tests (https://github.com/helm/helm/blob/master/docs/chart_tests.md) and verify if etcd cluster is _really_ started and alive.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)
- [x] Tested with Helm `v2.14.3`, K8s `v1.15.2`

### CODE REVIEW

**Comment by armab** [comment]
@lachie83 @alejandroEsc Can you please take a look at this PR?

**Comment by alejandroEsc** [comment]
@armab I need to take some time to review these changes, please be patient.

**Comment by armab** [comment]
@alejandroEsc  Absolutely. Happy to address any questions appearing along the way.

File: stable/etcd-operator/values.yaml [hunk 1]
```diff
@@ -154,4 +154,4 @@ etcdCluster:
     ## Ref: https://kubernetes.io/docs/user-guide/node-selection/
     nodeSelector: {}
     securityContext: {}
-    tolerations: {}
+    tolerations: []
```
Comment (arm4b) [inline stable/etcd-operator/values.yaml]: Extracted this fix into a dedicated PR: https://github.com/helm/charts/pull/17068

Comes from regression introduced in https://github.com/helm/charts/pull/16085

**Comment by alejandroEsc** [comment]
@armab can you please rebase given the latest merge. Just want to review again the changes you are making.

**Comment by armab** [comment]
@alejandroEsc Sure, synced up with latest in https://github.com/helm/charts/pull/16714/commits/647059e47048db131198d30b6ceaabbdbb011fef

**Comment by armab** [comment]
Thanks!

**Comment by alejandroEsc** [comment]
> Thanks!

No problem, thanks for bearing with me as I figure a few things out.
